### PR TITLE
Feat: Display Recommended Workflows in VSCode Plugin

### PR DESCRIPTION
**Pull request by yangbobo2021**
## **User description**
This pull request integrates the display of recommended workflows within the devchat VSCode plugin, addressing the user experience enhancement detailed in issue #305.

The following changes have been implemented:

1. Dependencies have been updated, adding @iarna/toml to `package.json` for the configuration file parsing.

2. The `fs` and `os` modules along with promisify have been imported to enable asynchronous file reading.

3. An asynchronous method has been created to load and parse the recommended workflows configuration file.

4. The `CommandEntry` interface has been extended, and the command logic has been adjusted to incorporate workflow recommendations.

This PR closes devchat-ai/devchat#305.

___

## **Type**
enhancement

___

## **Description**
- Added support for displaying recommended workflows in the devchat VSCode plugin.
- Implemented asynchronous loading and parsing of a TOML configuration file to fetch recommended workflows.
- Extended the `CommandEntry` interface to include a `recommand` field for ranking recommendations.
- Updated `package.json` to include `@iarna/toml` for TOML file parsing.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>devchat.ts</strong><dd><code>Integrate Recommended Workflows Display in DevChat Tool</code>&nbsp; &nbsp; </dd></summary>
<hr>
      
src/toolwrapper/devchat.ts

<li>Imported <code>fs</code>, <code>os</code>, and <code>util</code> modules for asynchronous file operations.<br> <li> Added <code>@iarna/toml</code> import for TOML file parsing.<br> <li> Implemented <code>loadRecommandCommands</code> async method to load and parse <br>recommended workflows.<br> <li> Modified <code>commands</code> method to include recommended command rankings based <br>on external configuration.<br>

</details>
    

  </td>
  <td><a href="https://github.com/devchat-ai/devchat-vscode/pull/469/files#diff-767083ebd7685a9d23cc75b6292b5bdaa1dd2a347f677ed41bb1342339c2d612">+59/-1</a>&nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Dependencies
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>package.json</strong><dd><code>Update Dependencies for Workflow Recommendations</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
package.json

- Added `@iarna/toml` dependency for TOML file parsing.

</details>
    

  </td>
  <td><a href="https://github.com/devchat-ai/devchat-vscode/pull/469/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: src/toolwrapper/devchat.ts [hunk 1]
```diff
@@ -425,7 +468,29 @@ class DevChat {
...
+			} catch (error) {
+				logger.channel()?.error('Failed to parse commands JSON:', error);
+				return [];
+			}
+			
+
+			// 确保每个CommandEntry对象的recommend字段默认为-1
+			commands = commands.map((cmd: CommandEntry) => ({
+				...cmd,
+				recommend: cmd.recommend ?? -1,
```
Comment (pplam) [inline src/toolwrapper/devchat.ts:483]: 直接在这里找到它在 recommendCommands 中的 index 加上就可以了，不用再去迭代一遍 recommendCommands

**Review by pplam — CHANGES_REQUESTED** [review]

File: src/toolwrapper/devchat.ts (hunk 1 above)
Comment (yangbobo2021) [reply src/toolwrapper/devchat.ts:483]: 放这里也是可以的，这样代码更加简洁。没有放这里的主要原因是，希望将逻辑分为两个更容易理解的步骤，步骤1，加载所有命令列表，步骤2，处理过滤命令列表序号。多几行代码，理解上会更简单一些，所以选择了这种方式。

Comment (yangbobo2021) [reply src/toolwrapper/devchat.ts]: 再多考虑了下，好像将步骤修改为这样也是合适的，步骤1，加载原始命令列表；步骤2，设置过滤排序值。按这个步骤设计，那确实应该把代码进行合并处理。

**Review by pplam — APPROVED** [review]

**Pull request merged**